### PR TITLE
fix broken mdx links

### PR DESCRIPTION
**Pull request by sag1v**
Issue: #16918 

## What I did
chained the hash to the location.search instead of just passing it to `navigate`

I think this is a critical bug fix and should get released ASAP :)

### CODE REVIEW

File: addons/docs/src/blocks/mdx.tsx [hunk 1]
```diff
@@ -49,8 +49,16 @@ export const CodeOrSourceMdx: FC<CodeOrSourceMdxProps> = ({ className, children,
   );
 };
 
+function generateHrefWithHash(hash: string): string {
+  // eslint-disable-next-line no-undef
+  const location = window.parent ? window.parent.location : window.location;
```
Comment (ndelangen) [inline addons/docs/src/blocks/mdx.tsx:54]: This is an illegal operation cross origin!

Comment (ndelangen) [reply addons/docs/src/blocks/mdx.tsx:54]: which can happen when a storybook is composed in:
https://storybook.js.org/docs/react/sharing/storybook-composition#gatsby-focus-wrapper

**Review by ndelangen — CHANGES_REQUESTED** [review]

File: addons/docs/src/blocks/mdx.tsx (hunk 1 above)
Comment (sag1v) [reply addons/docs/src/blocks/mdx.tsx:54]: @ndelangen  Got any suggestions to update the location?

Comment (sag1v) [reply addons/docs/src/blocks/mdx.tsx:54]: BTW, your link seems to point to a header called "gatsby-focus-wrapper" but i don't see it in the page

Comment (sag1v) [reply addons/docs/src/blocks/mdx.tsx:54]: I think that the issue is not the check but the fact that we're not passing the correct URL
We are passing only the hash and not appending it to the already existing URL or replacing it with existing hash in the URL.

Comment (sag1v) [reply addons/docs/src/blocks/mdx.tsx:54]: When we call `addons.getChannel().emit(NAVIGATE_URL, url)`, where is the actual code that navigating? Maybe we will have access to the full URL there and we can change the logic there.

There's a `getBase` function in `router.tsx`

```js
const getBase = () => `${document.location.pathname}?`;
```

What are the implications of `document.location` vs `window.location`?

**Comment by shilman** [comment]
Thanks @sag1v @ndelangen 

I bisected and confirmed that the code broke in `6.4.0-beta.20` which contains https://github.com/storybookjs/storybook/pull/16440 upgrading from `reach-router` to `react-router`

I believe the error is in this code:

https://github.com/storybookjs/storybook/pull/16440/files#diff-f62af13e39612cc8ebcb82a37a3a55afa29dbea92a0bd94162404a5e7e52aca1R136-R144

And that there's a missing test case which corresponds to `navigateTo('#some-hash')`. Do you think you could fix the bug on that side of things?

**Comment by sag1v** [comment]
@shilman I'm not sure i understand. I don't think we are even calling `navigateTo` in `url.ts`.
The only time i see we are calling this function is when we are switching from `Docs` tab to `Canvas` tab.

**Comment by shilman** [comment]
@sag1v You might be right. But it's definitely that PR that broke the behavior. Perhaps it's just how react-router handles navigation vs. reach-router. @ndelangen can you help out here?

**Comment by ndelangen** [comment]
@shilman @yannbf and me looked at this together and came up with a solution so the hash-url is just handled correctly at the router's level.

I opened a PR https://github.com/storybookjs/storybook/pull/17134

Really really appreciate your help @sag1v

**Comment by sag1v** [comment]
@ndelangen Thanks, i can't wait to test it in our project.

Just to add my 2 cent, i think [this is where](https://github.com/sag1v/storybook/commit/bec2df77c833001c466fe888bef52ed172baa5a3#diff-25fac9fce88adfd0db57f0d3a836b1d006b79c1e86b1413029621940456d62c5R158) things started to break

**Comment by shilman** [comment]
closing this as it's superseded by #17134

@sag1v thanks so much for pushing on this 🙏

**Comment by sag1v** [comment]
@shilman thank you!
Which version can I use to test it?

**Comment by shilman** [comment]
@sag1v 6.5.0-alpha.x

```
npx sb@next upgrade --prerelease
```

If you can confirm I'll patch it back and release it as part of 6.4.10

**Pull request closed without merging**